cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/http/HttpHandlerGenerator.java

```diff
@@ -501,8 +502,9 @@ public void visitClassType(String name) {
      *       txContext.finish();
      *     } catch (TransactionFailureException e) {
      *        LOG.error("Transaction failure: ", e);
-     *        wrapResponder(responder).sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+     *        wrappedResponder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
      *     }
+     *     ((DefaultHttpServiceResponder) wrappedResponder).execute();
```

Comment:
why not make wrapResponder() return DefaultHttpServiveResponder, then you don't need to cast here. Also, it seems that DefaultHttpServiceResponder is only used here, so why not call it a WrappedHttpServiceResponder, or DelayedHttpServiceResponder?